aws/rust-runtime/aws-config/src/imds/client.rs

```diff
@@ -237,7 +232,9 @@ impl Client {
         &self,
         path: &str,
     ) -> Result<Operation<ImdsGetResponseHandler, ImdsResponseRetryClassifier>, ImdsError> {
-        let mut base_uri: Uri = path.parse().map_err(|_| ImdsError::InvalidPath)?;
+        let mut base_uri: Uri = path.parse().map_err(|_| {
+            ImdsError::unexpected("IMDS path was not a valid URI. Hint: does it begin with `/`?")
```

Comment:
I think having a hint is good but it's not clear if the hint is saying that it should or should not begin with `/`

Reply:
```suggestion
            ImdsError::unexpected("IMDS path was not a valid URI. Hint: valid URIs begin with `/`")
```